### PR TITLE
Enhancements.

### PR DESCRIPTION
**Pull request by evanstade**
1. Specify when expired buckets are removed.

2. Improve some text around durability, quota.

3. Specify failure for invalid `quota`, `expiration` arguments to `open()`. Also, `quota` should not be updated for existing buckets.

***
<a href="https://pr-preview.s3.amazonaws.com/evanstade/storage-buckets/pull/56.html" title="Last updated on Jan 5, 2023, 11:03 PM UTC (305f454)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/storage-buckets/56/4c79005...evanstade:305f454.html" title="Last updated on Jan 5, 2023, 11:03 PM UTC (305f454)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -17,11 +17,14 @@ Markup Shorthands: css no, markdown yes
 <pre class=anchors>
 spec: storage; urlPrefix: https://storage.spec.whatwg.org/
     type: dfn
+        text: bottle map; url: bottle-map
         text: bucket map; url: bucket-map
         text: mode; url: bucket-mode
         text: obtain a local storage shelf; url: obtain-a-local-storage-shelf
         text: storage bucket; url: storage-bucket
+        text: storage bottle; url: storage-bottle
```
Comment (ayuishii) [inline index.bs:25]: super nit: if this is alphabetical, this should go above storage bucket

File: index.bs [hunk 2]
```diff
@@ -198,7 +215,13 @@ The <dfn method for="StorageBucketManager">keys()</dfn> method steps are:
 
     1. Let |keys| be a new [=/list=].
 
-    1. For each |key| in |shelf|'s [=bucket map=], [=list/append=] |key| to |keys|.
+    1. For each |key| in |shelf|'s [=bucket map=], run the following steps:
+
+        1. Let |bucket| be |shelf|'s [=/bottle map=][|key|].
```
Comment (ayuishii) [inline index.bs:220]: Should this be `[=/bucket map=][|key|]`, because I think we want the bucket that holds the bottle map?

**Review by ayuishii — COMMENTED** [review]
Thanks for filling in the gaps!

File: index.bs (hunk 1 above)
Comment (evanstade) [reply index.bs:25]: done

File: index.bs (hunk 2 above)
Comment (evanstade) [reply index.bs:220]: yes, done.

**Review by ayuishii — APPROVED** [review]
lgtm, thanks!

**Pull request merged**